packages/validator/src/util/externalSignerClient.ts

```diff
@@ -157,20 +164,34 @@ export async function externalSignerPostSignature(
 export async function externalSignerUpCheck(remoteUrl: string): Promise<boolean> {
   const res = await fetch(`${remoteUrl}/upcheck`, {
     method: "GET",
-    headers: {"Content-Type": "application/json"},
+    headers: {Accept: MediaType.json},
   });
 
-  const data = await handlerExternalSignerResponse<{status: string}>(res);
+  const data = await handleExternalSignerResponse<{status: string}>(res);
   return data.status === "OK";
 }
 
-async function handlerExternalSignerResponse<T>(res: Response): Promise<T> {
+async function handleExternalSignerResponse<T>(res: Response): Promise<T> {
   if (!res.ok) {
     const errBody = await res.text();
-    throw Error(`${errBody}`);
+    throw Error(errBody || res.statusText);
+  }
+
+  const contentType = res.headers.get("content-type");
+  if (contentType === null) {
+    throw Error("No Content-Type header found in response");
   }
 
-  return JSON.parse(await res.text()) as T;
+  const mediaType = contentType.split(";", 1)[0].trim().toLowerCase();
+  if (mediaType !== MediaType.json) {
+    throw Error(`Unsupported response media type: ${mediaType}`);
+  }
+
+  try {
+    return (await res.json()) as T;
```

Comment:
Looks like the spec [defines](https://github.com/ethereum/remote-signing-api/blob/7879ed3284bf88acb1149d30a70fbf8b554c183b/signing/paths/sign.yaml#L607) support for `text/plain` responses too? Are we fine not supporting it? (wasn't the case before either)

Reply:
We just implement the client side of the spec, there is no requirement to accept text responses as spec compliant server must support both response types and return the requested format in accept header. See my previous comment on this https://github.com/ChainSafe/lodestar/pull/6587#pullrequestreview-1977898563